### PR TITLE
Ports TG's visualized blushing when you use *blush or are drunk

### PR DESCRIPTION
**Pull request by Chubbygummibear**
# Document the changes in your pull request
*blush makes your spessperson blush, and getting drunk with mood enabled will make you blushy. if you have mood off being drunk won't do the blush

I know these are horribly cropped, i'm a coder not a photographer
imageimageimage
imageimageimage
And Yog's own snowflake race
imageimage

Ports 

- [66321](https://github.com/tgstation/tgstation/pull/66321) Visualizes blushing on characters after using *blush or when they're drunk [66321](https://github.com/tgstation/tgstation/pull/66321)
and
- [46325](https://github.com/tgstation/tgstation/pull/46325) Applies arglist memes to mood events. Also fixes some bugs

I know these should be atomized but i need the updated code to make this work so if people don't want the blushing i can just port the moodcode update later. also the last time base moodcode was touched was over a year ago so, sorry maintainers, i'm combo-ing. (i really don't want to wait a week for the mood code update to get merged and another week to do this)

# Spriting
Adds the blush sprite, it's like 4 white pixels just below the eyes

# Wiki Documentation

i don't think there's extensive emote documentation but maybe the drunk blushing is worth mentioning if you have mood on

# Changelog

Edit the changelog below to reflect the changes made by this PR, even if the changes are minor - required for every PR.
If you add a name after the ':cl', that name will be used in the changelog. Leave it empty to use your GitHub name.

:cl:  
rscadd: *Blush emote has a visual to go with it
rscadd: Being drunk with mood enabled will also make you blush. If you have mood off, you won't have to worry about accidentally showing some character expression
tweak: moodcode backend updates to support argslist and renaming the base mood event to _mood_event so it shows up first
imageadd: Blurshed sprite
/:cl:

### CODE REVIEW

File: code/__DEFINES/colors.dm [hunk 1]
```diff
@@ -26,6 +26,8 @@
 #define COLOR_BLUE_LIGHT       "#33CCFF"
 #define COLOR_NAVY             "#000080"
 #define COLOR_PINK             "#FFC0CB"
+#define COLOR_BLUSH_PINK       "#DE5D83"
+#define COLOR_BLUSH_POLYSMORPH "#00FF32"
```
Comment (tattax) [inline code/__DEFINES/colors.dm:30]:
```suggestion
#define COLOR_BLUSH_PINK       "#DE5D83"
#define COLOR_BLUSH_POLYSMORPH "#00FF32"
```
this feels out of place but it'd need further input tbh

you could define it on the place you use the colors instead

File: code/__DEFINES/traits.dm [hunk 2]
```diff
@@ -329,6 +329,8 @@
 #define FRENZY_TRAIT "frenzy_trait"
 #define HORROR_TRAIT "horror"
 #define HOLDER_TRAIT "holder_trait"
+/// This person is blushing
+#define TRAIT_BLUSHING "blushing"
```
Comment (tattax) [inline code/__DEFINES/traits.dm:333]:
```suggestion
#define TRAIT_BLUSHING "blushing" // for blushing
```
\
```suggestion
#define TRAIT_BLUSHING "blushing"
```

File: code/modules/mob/living/carbon/human/species.dm [hunk 3]
```diff
@@ -690,6 +690,17 @@ GLOBAL_LIST_EMPTY(mentor_races)
...
+		// blush
+		if (HAS_TRAIT(H, TRAIT_BLUSHING)) // Caused by either the *blush emote or the "drunk" mood event
+			if(ispolysmorph(H))
+				var/mutable_appearance/blush_overlay = mutable_appearance('icons/mob/human_face.dmi', "blush", -BODY_FRONT_LAYER) //polys need the blush above their dome
+				blush_overlay.color = COLOR_BLUSH_POLYSMORPH
+				standing += blush_overlay
+			else
+				var/mutable_appearance/blush_overlay = mutable_appearance('icons/mob/human_face.dmi', "blush", -BODY_ADJ_LAYER) //should appear behind the eyes
+				blush_overlay.color = COLOR_BLUSH_PINK
+				standing += blush_overlay
```
Comment (tattax) [inline code/modules/mob/living/carbon/human/species.dm:702]:
```suggestion
		if(HAS_TRAIT(H, TRAIT_BLUSHING)) // Caused by either the *blush emote or the "drunk" mood event
			var/mutable_appearance/blush_overlay = mutable_appearance('icons/mob/human_face.dmi', "blush", -BODY_ADJ_LAYER) //should appear behind the eyes
			blush_overlay.color = COLOR_BLUSH_PINK
			if(ispolysmorph(H))
				blush_overlay = mutable_appearance('icons/mob/human_face.dmi', "blush", -BODY_FRONT_LAYER) //polys need the blush above their dome
				blush_overlay.color = COLOR_BLUSH_POLYSMORPH
			standing += blush_overlay
```

File: code/datums/mood_events/generic_negative_events.dm [hunk 4]
```diff
@@ -92,13 +92,12 @@
...
-		if(iscatperson(H))
-			H.dna.species.start_wagging_tail(H)
-			addtimer(CALLBACK(H.dna.species, /datum/species.proc/stop_wagging_tail, H), 30)
-			description =  "<span class='nicegreen'>They want to play on the table!</span>\n"
-			mood_change = 2
+	if(iscatperson(owner))
+		var/mob/living/carbon/human/H = owner
+		H.dna.species.start_wagging_tail(H)
+		addtimer(CALLBACK(H.dna.species, /datum/species.proc/stop_wagging_tail, H), 3 SECONDS)
+		description =  "<span class='nicegreen'>They want to play on the table!</span>\n"
```
Comment (tattax) [inline code/datums/mood_events/generic_negative_events.dm:99]:
```suggestion
		description =  "[span_nicegreen("They want to play on the table!")]\n"
```

**Comment by SomeguyManperson** [comment]
no

**Comment by KillerOrcaCora** [comment]
Damn…No sprite review for me.

**Comment by SomeguyManperson** [comment]
I don't know why we removed the "why is this good for the game" part of the pr description but it's just 
why 
what would this improve
I can't really see this being used for anything outside cringe and it being an "indicator for drunk people" doesnt work because around 80% of the playerbase has mood off anyway

**Comment by truz** [comment]
It makes an emote visible which is good, also yes it is a drunk indicator, we already merged handholding and have a kiss emote, what does blushing being visible change thats so horrendous you had to close the pr. its an upgrade to the emote cause you can see it on characters

"most people have mood off so its not a drunk indicator" playerbase issue, it is still a drunk indicator for the people who care about rp and may get their character drunk in the first place.

you thought lizards licking their eyeballs apparently adds a lot to the game but on an actually useful pr that shows drunkness and makes an emote visible on characters you close it.

Also this pr is liked by the playerbase

**Comment by JamieD1** [comment]
Nope. 2nded Theos

**Comment by truz** [comment]
whats so bad about it being visible we already have the emote so it being visible wouldnt add much else. "people i dont like may use it" ok coders generally dont like felinids yet you gave them the ability to meow so whats wrong with blushing being seen

**Comment by gaminada** [comment]
For rp sake and for fun time having. I dont see how this is bad for the game.

**Comment by Sniblet** [comment]
headcuder sock

**Pull request closed without merging**